src/Generator/OrderItemUnitCreditMemoUnitGenerator.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\RefundPlugin\Generator;
+
+use Sylius\Component\Core\Model\AdjustmentInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\OrderItemUnitInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Sylius\RefundPlugin\Entity\CreditMemoUnit;
+use Sylius\RefundPlugin\Entity\CreditMemoUnitInterface;
+use Webmozart\Assert\Assert;
+
+final class OrderItemUnitCreditMemoUnitGenerator implements CreditMemoUnitGeneratorInterface
+{
+    /** @var RepositoryInterface */
+    private $orderItemUnitRepository;
+
+    public function __construct(RepositoryInterface $orderItemUnitRepository)
+    {
+        $this->orderItemUnitRepository = $orderItemUnitRepository;
+    }
+
+    public function generate(int $unitId): CreditMemoUnitInterface
+    {
+        /** @var OrderItemUnitInterface $orderItemUnit */
+        $orderItemUnit = $this->orderItemUnitRepository->find($unitId);
+        Assert::notNull($orderItemUnit);
+
+        /** @var OrderItemInterface $orderItem */
+        $orderItem = $orderItemUnit->getOrderItem();
+
+        $discount = 0;
```

Comment:
Getting and adding discounts in this class looks a little bit tricky to me. I would wonder if we need to list discounts on Credit Memo and if so maybe separating this logic to another service would be suitable.